File: backend/core/src/main/java/com/feed_grabber/core/rabbit/Sender.java
package com.feed_grabber.core.rabbit;

import com.feed_grabber.core.rabbit.entityExample.CloseRequest;
import com.feed_grabber.core.rabbit.entityExample.MailEntity;
import com.feed_grabber.core.rabbit.entityExample.MailType;
import org.springframework.beans.factory.annotation.Value;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.Date;
import java.util.UUID;

@Component
@Slf4j
public class Sender {
    @Value("${rabbitmq.exchange}")
    private String exchange;

    @Value("${rabbitmq.routing-key}")
    private String routingKey;

    @Value("${rabbitmq.routing-key-request-close}")
    private String requestCloseRoutingKey;

    private final RabbitTemplate template;

    @Autowired
    public Sender(RabbitTemplate template) {
        this.template = template;
    }

    public void sendToProcessor(String message, String email, String type) {
        log.info(" [x] Sending...");
        this.template.convertAndSend(exchange, routingKey, new MailEntity(MailType.valueOf(type), message, email));
        log.info(" [x] Sent '{}'", message);
    }

    public void sendReportCloseRequest(UUID requestId, Date closeDate) {
        log.info(" [x] Sending...");
        this.template.convertAndSend(exchange, requestCloseRoutingKey, new CloseRequest(requestId, closeDate));
        log.info(" [x] Sent message for closing request with id {} at {}", requestId, closeDate);
    }

}

File: backend/event_processor/src/main/kotlin/com/feed_grabber/event_processor/rabbit/Sender.kt
package com.feed_grabber.event_processor.rabbit

import com.feed_grabber.event_processor.report.dto.ReportFileCreationDto
import com.feed_grabber.event_processor.report.dto.ReportFilesResponseDto
import org.springframework.amqp.rabbit.core.RabbitTemplate

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

import java.util.UUID

@Component
class Sender {
    @Value("\${rabbitmq.exchange}")
    private val exchange: String? = null

    @Value("\${rabbitmq.routing-key-response}")
    private val routingKey: String? = null

    @Value("\${rabbitmq.routing-key-response-links}")
    private val linksResponseRoutingKey: String? = null

    @Value("\${rabbitmq.routing-key-request-close-receive}")
    private val closeRequestRoutingKey: String? = null

    @Autowired
    private val template: RabbitTemplate? = null

    fun send(text: String) {
        println(" [x] Sending...")
        template!!.convertAndSend(exchange!!, routingKey!!, text)
        println(" [x] Sent $text")
    }

    fun sendLinks(links: ReportFilesResponseDto) {
        println(" [x] Sending generated file links to backend")
        template!!.convertAndSend(exchange!!, linksResponseRoutingKey!!, links)
        println(" [x] Sent generated file links to backend")
    }

    fun closeRequestNow(requestId: UUID) {
        println(" [x] Sending requestId for closing request")
        template!!.convertAndSend(exchange!!, closeRequestRoutingKey!!, requestId)
        println(" [x] Sent requestId")
    }

}
